tinyexpr.c

```diff
@@ -113,6 +119,20 @@ void te_free(te_expr *n) {
 
 static double pi() {return 3.14159265358979323846;}
 static double e() {return 2.71828182845904523536;}
+static double fac(double a) {/* simplest version of fac */
+    if (a < 0.0)
+        return NAN;
+    if (a > UINT_MAX)
+        return INFINITY;
+    unsigned int ua = (unsigned int)(a);
+    unsigned long int result = 1, i = 1;
+    for (i = 1; i <= ua; i++) {
```

Comment:
Defining variables inside for loops isn't standard I think. The `i` is used only for multiplication with `result`, which means when compiled it may be optimized into a register, which has to be the same size as `result` in order to be able to multiply without casting. By having `i` being an int, you would still have the same long register for the multiplication, but you'd have an extra int for storing the value.